Provide tasklists with ckeditor-list and tasklist filter

oliverguenther opened the pull request:
- [x] Depends on release of ckeditor5-list
- [x] Styling issues due to global `label` styles
- [x] Feature test
- Out of scope: Editing checkboxes in non-editing state?

https://github.com/opf/commonmark-ckeditor-build/pull/39

oliverguenther committed:
Provide tasklists with ckeditor-list and tasklist filter

oliverguenther commented:
The update seems to break macro functionality. Looking into it

HDinger reviewed (changes requested):
I've been waiting for checklists quite a long time, so I am really happy that they are soon coming. Except for the macros that are not visible, there are two other things I noticed:

* The comments look strange in edit mode (see picture). After save, there are no checklists displayed at all but only a normal bullet list.
<img width="267" alt="Bildschirmfoto 2019-08-30 um 11 33 02" src="https://user-images.githubusercontent.com/7457313/64015406-c2dc7600-cb24-11e9-9676-171bb0bcc95b.png">
* The checklists state cannot be changed by clicking in the box, when you are not in the edit mode. I assume, that this is not trivial. So it is fine for me if we ignore this for now, but I stumbled over this. However, when having a checklist in a description, a click on the box does not trigger the edit mode. At least that should happen.

oliverguenther committed:
Use unsafe option for commonmarker to allow HTML

This was the default before the upgrade. We need  HTML for tables and
macros and use the sanitization filter to remove non-whitelisted HTML.

oliverguenther committed:
Fix styling of task lists

oliverguenther commented:
I've extracted the editable checkboxes into https://community.openproject.com/wp/30899 . I think in the long run it would be unnecessary because we're going to render the CKEditor at all times to avoid having to switching into an edit mode.
